CMakeLists.txt

```diff
@@ -30,6 +31,9 @@ set(CMAKE_CXX_FLAGS "${FLAGS}")
 if (EXTERNAL_HIGHFIVE)
   find_package(HighFive REQUIRED)
 endif()
+if (EXTERNAL_PYBIND11)
+  find_package(pybind11 REQUIRED CONFIG)
```

Comment:
should EXTERNAL_HIGHFIVE also have `CONFIG`?

I'm tempted to redo this like we have it in libsonata; ie the `EXTLIB_FROM_SUBMODULES` and then everything must be supplied.  Thoughts?

Reply:
:shrug: it's CMake, so I guess consistency would be nice for `CONFIG`. This just skips attempting to find a `Findpybind11` (or a permutation of capitalization).

In general, being consistent across our projects would be nice. So I'd be in favor of a `EXTLIB_FROM_SUBMODULES`.

Reply:
> This just skips attempting to find a Findpybind11 (or a permutation of capitalization).

My CMake-Fu isn't great - is the HighFive CMake similar enough to the pybind11 that it will work w/ `CONFIG` in all cases?

> So I'd be in favor of a EXTLIB_FROM_SUBMODULES.

The question then is whether `lexertl14` and `GSL_LITE` should be part of that; seems unlikely that other use those packages, and having `pybind11` and `HighFive` only behind `EXTLIB_FROM_SUBMODULES` seems weird.  I'm vacillating; maybe it's best to have what you proposed here.

Reply:
HighFive should also work with `CONFIG`, I don't think any sane person should have written `FindHighFive.cmake`.

`GSL_LITE` is actually the one that bugs me the most — because its headers get installed alongside MorphIO, which violates my sense of package integrity. That's why we tend to use the external HighFive for C++.

OTOH, one would have to add a Spack package for `gsl-lite`, which I'm currently not inclined to do. MorphIO is currently the only consumer, and I'd hope that we could move to `std::ranges` at some point soon™.

Reply:
This has been sitting too long, I'm going to go with what you have here.

Reply:
(sitting too long, waiting for me, to be clear)